src/OpenIddict.Core/Stores/IOpenIddictApplicationStore.cs

```diff
@@ -0,0 +1,72 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/openiddict/openiddict-core for more information concerning
+ * the license and the contributors participating to this project.
+ */
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenIddict {
+    /// <summary>
+    /// Provides methods allowing to manage the applications stored in a database.
+    /// </summary>
+    /// <typeparam name="TApplication">The type of the Application entity.</typeparam>
+    public interface IOpenIddictApplicationStore<TApplication> where TApplication : class {
+        /// <summary>
+        /// Creates a new application.
```

Comment:
No strong opinion, I used `Creates` because it's the terminology used by Identity.